docs_src/design/adr/system-management/001-docker-image-guidelines.md

```diff
@@ -0,0 +1,117 @@
+# Docker image guidelines
+
+## Status
+
+Pending
+
+## Context
+
+When deploying the EdgeX Docker containers some security measures are recommended to ensure the integrity of the software stack.
+
+## Decision
+
+When deploying Docker images, the following flags should be set for heightened security.
+
+- To avoid escalation of privileges each docker container should use the `no-new-privileges` option in their Docker compose file (example below). More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#security-configuration). This follows Rule #4 for Docker security found [here](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-4-add-no-new-privileges-flag).
```

Comment:
We've also had instances of device services using the `--privileged` flag in order to access DBus and/or actual hardware devices. It'd be nice if this was called out explicitly as not allowed. In addition, we should provide some guidelines as to how to properly allow hardware access (e.g. writing custom AppArmor rules) from within a container.

Reply:
@tonyespy   Jim White I believe commented on this earlier.  I will repeat what I said before.

no-new-privileges doesn't prevent one from running privileged containers.  What it prevents is tools like "sudo" from running in a container thus a going from user mode to root mode.  It also prevents capabilities defined on extended file attributes from being effected.  (For example, the capability to bind to a low-numbered port.).

Docker-compose v3 doesn't allow one to customize cgroups like the v2 did.  For example, you can no longer pass in device_cgroup_rules.  So if you are accessing a device that isn't in docker's built-in allow list, privileged mode is one way to allow device access to continue to function.  (Curious myself if this is the only way, or maybe we should downrev the docker-compose to v2... hmmm.)

Reply:
Thanks @bnevis-i. Two additional points...

 * The [device-bluetooth-c](https://github.com/edgexfoundry-holding/device-bluetooth-c) (still in holding) device service recommends running the service using `docker run --privileged`, and on systems where AppArmor is supported, includes an example of how to pass an AppArmor profile to docker run. Do you know if compose v2 also allowed AppArmor profiles to be specified?
 * Frankly, if we're going to suggest that someone run a service with `--privileged`, I'm not sure why we're even bothering to run the service in a container in the first place.

Reply:
@FraserG01, who did the device-bluebooth-c apparmor implementation, do you know if an apparmor profile can be used to override dockers default device cgroup restrictions?

Reply:
@brian-intel 

Are you up to adding a recommendation to prefer

```
security_opt: [ "apparmor:unconfined" ]
```

or a custom apparmor profile vs running `--privileged`

This doesn't negate the guidance regarding no-new-privileges, but it does add additional (and worthy) guidance to not use --privileged at all.